include/xtensor-signal/find_peaks.hpp

```diff
@@ -0,0 +1,725 @@
+                        if (x(i) < left_min)
+                        {
+                            left_min = x(i);
+                            left_bases(peak_nr) = i;
+                        }
+                        i--;
+                    }
+
+                    right_bases(peak_nr) = peak;
+                    i = right_bases(peak_nr);
+                    auto right_min = x(peak);
+                    while (i <= i_max && x(i) <= x(peak))
+                    {
+                        if (x(i) <= right_min)
+                        {
+                            right_min = x(i);
+                            right_bases(peak_nr) = i;
+                        }
+                        i++;
+                    }
+
+                    prominences(peak_nr) = x(peak) - std::max(left_min, right_min);
+                }
+
+                return std::make_tuple(prominences, left_bases, right_bases);
+            }
+
+            /**
+             * @brief Ensure argument `wlen` is of type `np.intp` and larger than 1.
+             * @returns The original `value` rounded up to an integer or -1 if `value` was None.
+             * @note Derived from https://github.com/scipy/scipy/blob/main/scipy/signal/_peak_finding.py
+             */
+            int arg_wlen_as_expected(std::optional<size_t> value)
+            {
+                //if the value is a none type
+                if(!value.has_value())
+                {
+                    return -1;
+                }
+                else
+                {
+                    //otherwise we have a number
+                    //could probably add a check for arithmatic type here
+                    if (1 < value.value())
+                    {
+                        return std::ceil(value.value());
+                    }
+                }
+
+                throw std::runtime_error("wlen must be greater thank 1");
+
+                return 0;
+            }
+
+            /**
+             * @brief Calculate the width of each peak in a signal.
+             * @details This function calculates the width of a peak in samples at a relative
+             *     distance to the peak's height and prominence.
+             * @param x A signal with peaks.
+             * @param peaks Indices of peaks in `x`.
+             * @param rel_height
+             *     Chooses the relative height at which the peak width is measured as a
+             *     percentage of its prominence. 1.0 calculates the width of the peak at
+             *     its lowest contour line while 0.5 evaluates at half the prominence
+             *     height. Must be at least 0. See notes for further explanation.
+             * @param prominence_data 
+             *     A tuple of three arrays matching the output of `peak_prominences` when
+             *     called with the same arguments `x` and `peaks`. This data are calculated
+             *     internally if not provided.
+             * @param wlen 
```

Comment:
This parameter does not match those of the function definition; `left_bases` and `right_bases` are not documented.